Client/Frontend/Home/ActivityStreamPanel.swift

```diff
@@ -456,26 +475,25 @@ extension ActivityStreamPanel {
             self.presentViewController(controller, animated: true, completion: nil)
         })
 
-        let removeTopSiteAction = ActionOverlayTableViewAction(title: Strings.RemoveFromASContextMenuTitle, iconString: "action_close", handler: { action in
+        let removeTopSiteAction = ActionOverlayTableViewAction(title: Strings.RemoveContextMenuTitle, iconString: "action_remove", handler: { action in
             ASOnyxPing.reportDeleteItemEvent(eventInfo)
             self.hideURLFromTopSites(site.tileURL)
         })
-        
-        let dismissHighlightAction = ActionOverlayTableViewAction(title: Strings.RemoveFromASContextMenuTitle, iconString: "action_close", handler: { action in
+
+        let dismissHighlightAction = ActionOverlayTableViewAction(title: Strings.RemoveContextMenuTitle, iconString: "action_remove", handler: { action in
             ASOnyxPing.reportDeleteItemEvent(eventInfo)
             self.hideFromHighlights(site)
         })
 
-        var actions = [openInNewTabAction, openInNewPrivateTabAction, bookmarkAction, shareAction]
-        switch eventInfo.source {
-            case .highlights: actions.appendContentsOf([dismissHighlightAction, deleteFromHistoryAction])
-            case .topSites: actions.append(removeTopSiteAction)
-            default: break
+        actions.appendContentsOf([bookmarkAction, shareAction])
+
+        switch Section(indexPath.section) {
```

Comment:
@bkmunar, I'm sorry I didn't understand this approach. Did you mean something like this?
```

var dismissHighlightAction, deleteFromHistoryAction, removeTopSiteAction: ActionOverlayTableViewAction

switch Section(indexPath.section) {
case .Highlights:
    // ...
    extraActions = [dismissHighlightAction, deleteFromHistoryAction]
case .TopSites:
    // ...
    extraActions = [removeTopSiteAction]
    // ...
}

// ...
actions.appendContentsOf([bookmarkAction, shareAction])
actions.appendContentsOf(extraActions)

```

This approach doesn't work because the compiler needs the actions to be initialized before adding them in the switch statement.